graph.cpp

```diff
@@ -0,0 +1,194 @@
+#include "graph.hpp"
+#include<vector>
+#include<iostream>
+#include<queue>
+
+void Graph::Node::insertIncoming(int64 from) {
+  incoming->insert(from);
+}
+
+void Graph::Node::insertOutgoing(int64 to) {
+  outgoing->insert(to);
+}
+
+void Graph::Node::eraseIncoming(int64 from) {
+  incoming->erase(from);
+}
+
+void Graph::Node::eraseOutgoing(int64 to) {
+  outgoing->erase(to);
+}
+
+bool Graph::Node::containsEdgeTo(int64 to) {
+  // apparently the standard c++ way to check if something is in range
+  // call this too many times to not make this a method...
+  return outgoing->find(to) != outgoing->end();
+}
+
+bool Graph::Node::containsEdgeFrom(int64 from) {
+  return incoming->find(from) != incoming->end();
+}
+
+bool Graph::Contains(int64 nodeid) {
+  return nodemap.find(nodeid) != nodemap.end();
+}
+
+Graph::Node Graph::Node::DeepCopy() {
+  // create a new node
+  auto result = Node();
```

Comment:
Cool, I can fix that. I tried using auto only where it was clear what the variable is being set to, but your reasoning is fair enough.